src/olympia/yara/tests/test_models.py

```diff
@@ -0,0 +1,33 @@
+from olympia.amo.tests import TestCase, addon_factory
+
+from olympia.files.models import FileUpload
+from olympia.yara.models import YaraResult
+
+
+class TestYaraResult(TestCase):
+    def test_create(self):
+        addon = addon_factory()
+        upload = FileUpload.objects.create(addon=addon)
+
+        result = YaraResult.objects.create(upload=upload)
+
+        assert result.id is not None
+        assert result.upload == upload
+        assert result.matches == []
+        assert result.version is None
+
+    def test_add_match(self):
+        addon = addon_factory()
+        upload = FileUpload.objects.create(addon=addon)
+        result = YaraResult.objects.create(upload=upload)
+
+        rule = 'some-yara-rule'
+        tags = ['foo']
+        meta = {'description': 'some description for some-yara-rule'}
+        result.add_match(rule=rule, tags=tags, meta=meta)
+
+        assert result.matches == [{
+            'rule': rule,
+            'tags': tags,
+            'meta': meta,
+        }]
```

Comment:
You could add a simple test creating a yararesult with a fileupload attached and then delete the fileupload and make sure the yararesult is still here, proving you've set `on_delete` correctly.

Reply:
just added that